APE18.rst

```diff
@@ -0,0 +1,59 @@
+Adopt NEP 29 for CPython and Numpy Version Support
+--------------------------------------------------
+
+author: Stuart Mumford and Thomas Robitaille
+
+date-created: 2020 October 16
+
+date-last-revised: 2020 October 16
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+APE 10 previously determined supported CPython versions, and provided a list up to the 4.0 release.
+Since APE 10 was written the wider scientific Python community have written and largely adopted `NEP 29`_ as a policy for supported versions of both CPython and Numpy.
+This APE proposes that Astropy adopt this policy following other projects in the scientific ecosystem.
+
+
+Detailed description
+--------------------
+
+This APE fully adopts `NEP 29`_ for both CPython and Numpy versions, it supersedes APE 10 wherever there is a conflict between that APE and this APE or `NEP 29`_.
```

Comment:
I question raised by @bsipocz is should we allow, or even mandate, that an LTS release can jump forward a numpy (or even CPython?!) version to reduce the long term maintenance burden.

I think a good way to do this would be to say something like "An LTS release of the core package may set the minimum versions for the next release" (i.e. 5.0 could adopt the versions specified by 5.1).

Reply:
I would just say that institutions that are using the LTS are more likely going to have a stack that has generally older versions of Python and dependencies. So the target audience for LTS may not be happy about this.

E.g. Chandra operations is currently using Python 3.6 + numpy 1.12. We're working on an upgrade now, but it turns out to be a lot of work in our setting (review board for any software changes, etc).

Reply:
i would just stick to the NEP 29 rules - if we had done for 4.0 I believe we would have required Numpy 1.15? Not too different from 1.16.

Reply:
The fact that it was bumped for 4.0 is the reason for this thread. That decision was clearly motivated by something. @mhvk ?

Reply:
Yes, we bumped to numpy 1.16 since then we could support quantities on nearly all numpy functions, i.e., there was what was considered a very compelling reason (and LTS played a role, as with the bump the user experience would not depend on numpy version; IIRC, it also was the case that in our old scheme it would have been OK just a few months later). 

Of course, generally numpy versions do not introduce major changes that are as directly relevant to astropy, so the general rule should definitely be NEP 29. But obviously there can be some room for exceptions - I guess it would be an argued addendum to this APE perhaps, i.e., a high bar.

Reply:
I would not mandate but allow. As I recall for 4.0, the timeline was just a few months early,   so it clearly made a sense both for the Quantity support, but also for planning ahead, to not to deal with 1.16 issues 2 years down the line

Reply:
Yes having a clause about allowing it is very specific cases makes sense - and doesn't have to apply to just LTS eg if 4.3 came out a few days before the next bump.

Reply:
I have added some words. If people want to review it.